### PR TITLE
Twitter not double tweet

### PR DESCRIPTION
**Pull request by shalecraig**
Re-rolling #400 and #401 

Noticed this bug. Fixed this bug. https://twitter.com/anythingbot

It's kindof bad that we duplicate this logic everywhere, but `¯\_(ツ)_/¯`

### CODE REVIEW

**Comment by botwillacceptanything** [comment]
#### :ballot_box_with_check: Voting procedure reminder:
To cast a vote, post a comment containing :+1:`:+1:`, or :-1:`:-1:`.
Remember, you **must :star:star this repo for your vote to count.**

All comments within this discussion are searched for votes, regardless of the time of posting.
You can cast as many votes as you want, but only the last one will be counted.
(You may consider editing your comment instead of adding a new one.)
Comments containing both :+1:up- and :-1:down-votes are disregarded.
Comments containing monkeys are disregarded. :monkey: :monkey_face: :see_no_evil: :hear_no_evil: :speak_no_evil:
PR authors automatically count as a :+1: vote.

A decision will be made after this PR has been open for **15** minutes (plus/minus **10** percent, to avoid people timing their votes), and at least **8** votes have been made.
A supermajority of **65%** is required for the vote to pass.

*NOTE: the PR will be closed if any new commits are added after:*
 38227403fdcf9bd22d7a761a428f76b700c7a0f2

**Comment by botwillacceptanything** [comment]
:-1:

**Comment by SpenserJ** [comment]
:+1:

File: tests/unit/twitter.js [hunk 1]
```diff
@@ -4,13 +4,14 @@
   var deps = [
     'assert',
     'nock',
+    'underscore',
```
Comment (avdg) [inline tests/unit/twitter.js]: Should be lodash?

Comment (shalecraig) [inline tests/unit/twitter.js]: Haven't heard of lodash. What do we use?

On Saturday, April 18, 2015, Anthony Van de Gejuchte <
notifications@github.com> wrote:

> In tests/unit/twitter.js
> <https://github.com/botwillacceptanything/botwillacceptanything/pull/403#discussion_r28647644>
> :
>
> > @@ -4,13 +4,14 @@
> >    var deps = [
> >      'assert',
> >      'nock',
> > +    'underscore',
>
> Should be lodash?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/botwillacceptanything/botwillacceptanything/pull/403/files#r28647644>
> .
>


-- 
----------------------------------------------------------------
Shale Craig
shalecraig@gmail.com

**Comment by avdg** [comment]
:-1: Not ready to merge, sorry

File: tests/unit/twitter.js (hunk 1 above)
Comment (avdg) [inline tests/unit/twitter.js]: lodash is actually like low-dash, its kinda the same as underscore, just a different library.

Comment (avdg) [inline tests/unit/twitter.js]: There is a lodash in package.js here https://github.com/botwillacceptanything/botwillacceptanything/blob/3142bc62204b88970b03e23db6db4fc8b7ea4df0/package.json#L21

But no underscore, making the ci-build fail.

Comment (avdg) [inline tests/unit/twitter.js]: Anyway, just replace `'underscore'` with `'lodash'` and it will be fine.

Docs are here https://lodash.com/docs#assign

**Comment by SpenserJ** [comment]
Good catch @avdg, :-1: until it uses lodash

**Comment by dos1** [comment]
:-1:

**Comment by timweightman** [comment]
:-1: Agree with others. lodash ftw.

**Comment by shalecraig** [comment]
:-1: can authors even vote? will re-roll. sorryguys

**Comment by dos1** [comment]
@shalecraig: yup, they can. They can fail the voting instantly by closing PR or adding new commits to it as well, just like you did now ;)

**Pull request closed without merging**